src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/ClusterToolMonitor.scala

```diff
@@ -0,0 +1,82 @@
+package org.broadinstitute.dsde.workbench.leonardo.monitor
+
+import akka.actor.{Actor, Props, Timers}
+import com.typesafe.scalalogging.LazyLogging
+import org.broadinstitute.dsde.workbench.google.GoogleProjectDAO
+import org.broadinstitute.dsde.workbench.leonardo.config.ClusterServiceConfig
+import org.broadinstitute.dsde.workbench.leonardo.dao.ToolDAO
+import org.broadinstitute.dsde.workbench.leonardo.dao.google.GoogleDataprocDAO
+import org.broadinstitute.dsde.workbench.leonardo.db.DbReference
+import org.broadinstitute.dsde.workbench.leonardo.model.{Cluster, ClusterTool}
+import org.broadinstitute.dsde.workbench.leonardo.monitor.ClusterToolMonitor.{DetectClusterStatus, TimerKey, ToolStatus}
+import org.broadinstitute.dsde.workbench.newrelic.NewRelicMetrics
+import cats.implicits._
+import org.broadinstitute.dsde.workbench.leonardo.model.ClusterTool.Welder
+
+import scala.concurrent.Future
+
+object ClusterToolMonitor {
+
+  def props(config: ClusterServiceConfig, gdDAO: GoogleDataprocDAO, googleProjectDAO: GoogleProjectDAO, dbRef: DbReference, toolDAOs: Map[ClusterTool, ToolDAO], newRelic: NewRelicMetrics): Props = {
+    Props(new ClusterToolMonitor(config, gdDAO, googleProjectDAO, dbRef, toolDAOs, newRelic))
+  }
+
+  sealed trait ClusterServiceMonitorMessage
+  case object DetectClusterStatus extends ClusterServiceMonitorMessage
+  case object TimerKey extends ClusterServiceMonitorMessage
+
+  case class ToolStatus(val isUp: Boolean, val tool: ClusterTool)
```

Comment:
It might be good to record the cluster project/name in `ToolStatus`. Otherwise we'll just log "welder is down" without knowing which cluster, which is probably useful information. :)